src/api/cpp/backend/backend_engine.h

```diff
@@ -26,87 +27,87 @@
 class nixlBackendEngine {
     private:
         // Members that cannot be modified by a child backend and parent bookkeep
-        nixl_backend_t   backendType;
-        nixl_b_params_t* customParams;
+        nixl_backend_t  backendType;
+        nixl_b_params_t customParams;
 
     protected:
         // Members that can be accessed by the child (localAgent cannot be modified)
-        bool              initErr;
+        bool              initErr = false;
         const std::string localAgent;
 
-        nixl_status_t setInitParam(const std::string &key, const std::string &value) {
-            if (customParams->count(key)==0) {
-                (*customParams)[key] = value;
+        [[nodiscard]] nixl_status_t setInitParam(const std::string &key, const std::string &value) {
+	    if (customParams.try_emplace(key,value).second) {
                 return NIXL_SUCCESS;
-            } else {
-                return NIXL_ERR_NOT_ALLOWED;
             }
+	    return NIXL_ERR_NOT_ALLOWED;
         }
 
-        nixl_status_t getInitParam(const std::string &key, std::string &value) {
-            if (customParams->count(key)==0) {
-                return NIXL_ERR_INVALID_PARAM;
-            } else {
-                value = (*customParams)[key];
+        [[nodiscard]] nixl_status_t getInitParam(const std::string &key, std::string &value) const {
+	    const auto iter = customParams.find(key);
+            if (iter != customParams.end()) {
+                value = iter->second;
                 return NIXL_SUCCESS;
-            }
+	    }
+	    return NIXL_ERR_INVALID_PARAM;
         }
 
     public:
-        nixlBackendEngine (const nixlBackendInitParams* init_params)
-            : localAgent(init_params->localAgent) {
-
-            this->backendType  = init_params->type;
-            this->initErr      = false;
-            this->customParams = new nixl_b_params_t(*(init_params->customParams));
+        explicit nixlBackendEngine (const nixlBackendInitParams* init_params)
+            : backendType( init_params->type ),
+              customParams( *init_params->customParams ),
+              localAgent(init_params->localAgent) {
```

Comment:
seems like different style is used in the project, without extra spaces in the brackets